centrallix/expression/exp_evaluate.c

```diff
@@ -320,7 +309,11 @@ expEvalDivide(pExpression tree, pParamObjects objlist)
 			    mssError(1,"EXP","Attempted divide by zero");
 			    return -1;
 			    }
-			mv = ((long long)(i0->Types.Money.WholePart)) * 10000 + i0->Types.Money.FractionPart;
+			md = i0->Types.Money.Value / 10000.0;
+			md = md / i1->Types.Double;
+			tree->Types.Money.Value = (long long)(md * 10000);
+			//Old Definition
+			/*mv = ((long long)(i0->Types.Money.WholePart)) * 10000 + i0->Types.Money.FractionPart;
```

Comment:
Looking at this again...I think my initial comment was wrong and your old code DID handle remainders (since it's dividing i0->Types.Money.Value / 10000.0 into a double type, so it should end up with a double with decimal places, etc). It seems like the adding/subtracting 0.5 might actually be related to the old representation's handling of negative values? (See also the adding 10000 to mv and subtracting one from WholePart on line 333 in the old version.) So you probably don't need to add/subtract 0.5?? Does that make sense to you?

Reply:
Yeah it makes sense.  I'll comment it out for now and see how it goes once I get to expression testing.

Reply:
So, I can omit the commented out section, and I also should go and simplify the code where it uses is_negative in this function?